lib/src/MixedHistogramUserDefined.cxx

```diff
@@ -239,15 +239,16 @@ OT::Distribution MixedHistogramUserDefined::getMarginal(const OT::UnsignedIntege
   const OT::UnsignedInteger dimension = getDimension();
   if (i >= dimension) throw OT::InvalidArgumentException(HERE) << "The index of a marginal distribution must be in the range [0, dim-1]";
   if (dimension == 1) return clone();
-  if (kind_[i] == 0)
-  {
-    const OT::UnsignedInteger size = ticksCollection_[i].getSize();
-    OT::SampleImplementation support(size, 1);
-    support.setData(ticksCollection_[i]);
-    OT::UserDefined marginal(support, mixture_.getMarginal(i).getProbabilities());
```

Comment:
I cannot check it right now, but:
```
OT::UserDefined marginal(mixture_.getMarginal(i).getSupport(), mixture_.getMarginal(i).getProbabilities());
```
should work here (don't duplicate mixture_.getMarginal(i) but you see the idea)

Reply:
Ok, that's another solution!